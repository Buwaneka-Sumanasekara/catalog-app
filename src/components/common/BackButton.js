import * as React from 'react';
import { Badge, IconButton, MD2Colors, useTheme } from 'react-native-paper';
import { useSelector } from 'react-redux';
import { StyleSheet, View } from 'react-native';

/* Styles ==================================================================== */
const styles = StyleSheet.create({
  iconStyle: {
    backgroundColor: MD2Colors.grey600,
    opacity: 0.9,
  },
});

const BackButton = (props) => {
  const { onPress } = props;
  return (
    <IconButton
      size={35}
      iconColor={MD2Colors.white}
      icon={'keyboard-backspace'}
      style={styles.iconStyle}
      onPress={onPress}
    />
  );
};

export default BackButton;
