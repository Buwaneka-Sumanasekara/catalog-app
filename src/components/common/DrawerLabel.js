import {StyleSheet, View} from "react-native"
import { IconButton, List } from "react-native-paper"

/* Styles ==================================================================== */
const styles = StyleSheet.create({
    containerStyle: {
        paddingVertical:0,
        paddingHorizontal:0,
        justifyContent:"space-between",
        flex:1,
        marginRight:-30,
    },
    iconStyle:{
        backgroundColor:'transparent'
    }
})

const DrawerLabel = (props) => {
    const {rightElement:RightElement,title}=props;
    return (
        <List.Item
        title={title}
        style={styles.containerStyle}
        left={props => <List.Icon {...props} icon="camera" />}
        right={RightElement?props=><RightElement {...props}/>:undefined}
        />
    )
}

export default DrawerLabel
