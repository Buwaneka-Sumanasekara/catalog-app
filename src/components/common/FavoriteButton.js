import { useCallback } from 'react';
import { IconButton, MD2Colors } from 'react-native-paper';
import { useDispatch } from 'react-redux';
import { StyleSheet } from 'react-native';
import ReduxActionTypes from '../../constants/ReduxActionTypes';

/* Styles ==================================================================== */
const styles = StyleSheet.create({});

export default FavoriteButton = (props) => {
  const { item, size, color } = props;

  const dispatch = useDispatch();

  const onPressFavorite = useCallback(
    (isRemove = false) => {
      if (isRemove) {
        dispatch({ type: ReduxActionTypes.REMOVE_FROM_FAVORITES, item: JSON.stringify(item) });
      } else {
        dispatch({ type: ReduxActionTypes.ADD_TO_FAVORITES, item: JSON.stringify(item) });
      }
    },
    [dispatch],
  );

  return (
    <IconButton
      size={size || 32}
      iconColor={color || MD2Colors.white}
      icon={item?.isFavorite ? 'cards-heart' : 'cards-heart-outline'}
      onPress={() => onPressFavorite(item?.isFavorite)}
    />
  );
};
