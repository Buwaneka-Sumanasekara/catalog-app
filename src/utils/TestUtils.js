import React from 'react';
import { render } from '@testing-library/react-native';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { QueryClient, QueryClientProvider } from 'react-query';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { NavigationContainer } from '@react-navigation/native';

const middlewares = [thunk];
const store = configureStore(middlewares);
const initialState = {
  favorites: {
    items: [],
    count: 0,
  },
};

const queryClient = new QueryClient();

const AllTheProviders = ({ children }) => {
  return (
    <Provider store={store(initialState)}>
      <QueryClientProvider client={queryClient}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </QueryClientProvider>
    </Provider>
  );
};

const customRender = (children) => render(<AllTheProviders>{children}</AllTheProviders>);

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };
