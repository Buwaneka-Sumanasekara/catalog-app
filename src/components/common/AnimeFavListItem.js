import { ImageBackground, Pressable, StyleSheet, View } from 'react-native';
import { Text, MD2Colors, Chip, IconButton } from 'react-native-paper';
import LinearGradient from 'react-native-linear-gradient';
import Globals from '../../constants/Globals';
import AppSizes from '../../constants/AppSizes';
import * as CommonUtil from '../../utils/CommonUtil';

/* Styles ==================================================================== */

const boxWidth = (AppSizes.screen.width - Globals.BOARDER_RADIUS.MEDIUM * 3) * 0.5;

const styles = StyleSheet.create({
  containerStyle: {
    width: boxWidth,
  },
  imageContainerStyle: {
    borderRadius: Globals.BOARDER_RADIUS.MEDIUM,
    overflow: 'hidden',
    marginBottom: Globals.PADDING.EXTRA_SMALL,
    minHeight: 200,
  },
  titleStyle: {
    color: MD2Colors.white,
  },
  imageStyle: {
    flex: 1,
  },
  linearGradientStyle: {
    flex: 1,
    opacity: 0.5,
    justifyContent: 'space-around',
  },
  imageInnerTopStyle: {
    position: 'absolute',
    top: Globals.PADDING.SMALL,
    zIndex: 10000,
    width: '100%',
    paddingHorizontal: Globals.PADDING.MEDIUM,
  },
  topChipStyle: {
    backgroundColor: MD2Colors.white,
    alignSelf: 'flex-end',
  },
  imageInnerBottomStyle: {
    position: 'absolute',
    bottom: Globals.PADDING.SMALL,
    right: Globals.PADDING.SMALL,
    left: Globals.PADDING.SMALL,
    zIndex: 10000,
    flexDirection: 'row',
    alignItems: 'center',
    width: '100%',
    justifyContent: 'space-between',
  },
  imageInnerBottomSingleItemStyle: {
    justifyContent: 'flex-end',
  },
  chipStyle: {
    backgroundColor: MD2Colors.white,
    marginLeft: Globals.PADDING.SMALL,
  },

  footerContainerStyle: {
    position: 'absolute',
    zIndex: 10000,
    bottom: Globals.PADDING.MEDIUM,
    left: Globals.PADDING.MEDIUM,
    width: boxWidth - Globals.PADDING.MEDIUM * 2,
  },
  footerChipStyle: {
    backgroundColor: MD2Colors.red100,
    paddingRight: 0,
  },
  yearContainerStyle: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-end',
    flex: 0.5,
  },
  yearIconStyle: {
    marginRight: -4,
  },

  innerContainerStyle: {
    flex: 1,
    justifyContent: 'center',
  },
});

const AnimeFavListItem = (props) => {
  const { index, columns, images, title, score, onPress } = props;

  const paddings = CommonUtil.getPaddings(columns || 1, Globals.PADDING.SMALL, index || 0);
  return (
    <Pressable style={[styles.containerStyle, paddings]} onPress={onPress}>
      <View style={styles.innerContainerStyle}>
        <View style={styles.imageContainerStyle}>
          <ImageBackground
            source={{ uri: images?.webp?.image_url }}
            resizeMode="cover"
            style={styles.imageStyle}
          >
            <View style={styles.imageInnerTopStyle}>
              {score ? (
                <Chip style={styles.topChipStyle} selected={true} icon="star">{`${score}`}</Chip>
              ) : null}
            </View>

            <View style={styles.footerContainerStyle}>
              <Text
                variant="titleMedium"
                numberOfLines={1}
                style={styles.titleStyle}
                ellipsizeMode={'tail'}
              >
                {title}
              </Text>
            </View>

            <LinearGradient
              colors={[MD2Colors.transparent, MD2Colors.black]}
              style={styles.linearGradientStyle}
            />
          </ImageBackground>
        </View>
      </View>
    </Pressable>
  );
};

export default AnimeFavListItem;
