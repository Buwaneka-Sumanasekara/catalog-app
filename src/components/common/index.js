import AnimeFavListItem from './AnimeFavListItem';
import AnimeListItem from './AnimeListItem';
import BackButton from './BackButton';
import EmptyView from './EmptyView';
import FavoriteButton from './FavoriteButton';
import FavoriteCount from './FavoriteCount';
import MenuButton from './MenuButton';
import SearchBarComponent from './SearchBarComponent';
import DrawerLabel from './DrawerLabel';

export {
  AnimeFavListItem,
  AnimeListItem,
  BackButton,
  EmptyView,
  FavoriteButton,
  FavoriteCount,
  MenuButton,
  SearchBarComponent,
  DrawerLabel,
};
