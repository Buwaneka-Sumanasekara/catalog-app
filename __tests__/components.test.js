import {AnimeFavListItem,EmptyView,DrawerLabel,SearchBarComponent,FavoriteButton,FavoriteCount,MenuButton} from '../src/components/common';
import { render } from '../src/utils/TestUtils'


describe('All components snapshot test:', () => {
  it('AnimeFavListItem snapshot:', () => {
      const { toJSON } = render(<AnimeFavListItem index={0} column={2} title={"test"}  onPress={()=>{}}/>)
      expect(toJSON()).toMatchSnapshot();
  })
  it('EmptyView snapshot:', () => {
    const { toJSON } = render(<EmptyView />)
    expect(toJSON()).toMatchSnapshot();
  })
  it('DrawerLabel snapshot:', () => {
    const { toJSON } = render(<DrawerLabel />)
    expect(toJSON()).toMatchSnapshot();
  })
  it('SearchBarComponent snapshot:', () => {
    const { toJSON } = render(<SearchBarComponent onChangeText={()=>{}} />)
    expect(toJSON()).toMatchSnapshot();
  })
  it('FavoriteButton snapshot:', () => {
    const { toJSON } = render(<FavoriteButton  />)
    expect(toJSON()).toMatchSnapshot();
  })
  it('FavoriteCount snapshot:', () => {
    const { toJSON } = render(<FavoriteCount  />)
    expect(toJSON()).toMatchSnapshot();
  })
  it('MenuButton snapshot:', () => {
    const { toJSON } = render(<MenuButton  />)
    expect(toJSON()).toMatchSnapshot();
  })
})
