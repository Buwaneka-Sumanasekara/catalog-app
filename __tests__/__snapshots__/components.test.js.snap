// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components snapshot test: AnimeFavListItem snapshot: 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "width": 360,
      },
      {
        "paddingBottom": 4,
        "paddingLeft": 0,
        "paddingRight": 4,
        "paddingTop": 8,
      },
    ]
  }
>
  <View
    style={
      {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        {
          "borderRadius": 10,
          "marginBottom": 6,
          "minHeight": 200,
          "overflow": "hidden",
        }
      }
    >
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          {
            "flex": 1,
          }
        }
      >
        <Image
          resizeMode="cover"
          source={
            {
              "uri": undefined,
            }
          }
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": undefined,
                "width": undefined,
              },
              undefined,
            ]
          }
        />
        <View
          style={
            {
              "paddingHorizontal": 16,
              "position": "absolute",
              "top": 8,
              "width": "100%",
              "zIndex": 10000,
            }
          }
        />
        <View
          style={
            {
              "bottom": 16,
              "left": 16,
              "position": "absolute",
              "width": 328,
              "zIndex": 10000,
            }
          }
        >
          <Text
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              [
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "letterSpacing": 0.15,
                  "lineHeight": 24,
                },
                {
                  "textAlign": "left",
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "#ffffff",
                },
              ]
            }
          >
            test
          </Text>
        </View>
        <LinearGradient
          colors={
            [
              "rgba(255, 255, 255, 0)",
              "#000000",
            ]
          }
          style={
            {
              "flex": 1,
              "justifyContent": "space-around",
              "opacity": 0.5,
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;

exports[`All components snapshot test: DrawerLabel snapshot: 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        {
          "flex": 1,
          "justifyContent": "space-between",
          "marginRight": -30,
          "paddingHorizontal": 0,
          "paddingVertical": 0,
        },
      ],
    ]
  }
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 6,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "alignSelf": "center",
            "marginLeft": 16,
            "marginRight": 0,
            "marginVertical": 0,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flex": 1,
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`All components snapshot test: EmptyView snapshot: 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "end": undefined,
        "flex": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "start": undefined,
        "top": undefined,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "flex": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container-inner-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": undefined,
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 33,
            "borderWidth": 0,
            "elevation": 0,
            "flex": undefined,
            "height": 66,
            "margin": 6,
            "overflow": "hidden",
            "width": 66,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 33,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 50,
                },
                [
                  {
                    "lineHeight": 50,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰋽
          </Text>
        </View>
      </View>
    </View>
  </View>
  <Text
    style={
      [
        {
          "textAlign": "left",
        },
        {
          "color": "rgba(28, 27, 31, 1)",
          "fontFamily": "System",
          "fontWeight": "400",
          "letterSpacing": 0,
        },
        {
          "writingDirection": "ltr",
        },
        undefined,
      ]
    }
  >
    No Data found
  </Text>
</View>
`;

exports[`All components snapshot test: FavoriteButton snapshot: 1`] = `
<View
  collapsable={false}
  style={
    {
      "alignSelf": undefined,
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "left": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "start": undefined,
      "top": undefined,
    }
  }
  testID="icon-button-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "flex": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="icon-button-container-inner-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": undefined,
          "borderColor": "rgba(121, 116, 126, 1)",
          "borderRadius": 24,
          "borderWidth": 0,
          "elevation": 0,
          "flex": undefined,
          "height": 48,
          "margin": 6,
          "overflow": "hidden",
          "width": 48,
        }
      }
      testID="icon-button-container"
    >
      <View
        accessibilityComponentType="button"
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityTraits="button"
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "overflow": "hidden",
            },
            false,
            [
              {
                "alignItems": "center",
                "flexGrow": 1,
                "justifyContent": "center",
              },
              {
                "borderRadius": 24,
              },
            ],
          ]
        }
        testID="icon-button"
      >
        <Text
          accessibilityElementsHidden={true}
          allowFontScaling={false}
          importantForAccessibility="no-hide-descendants"
          pointerEvents="none"
          selectable={false}
          style={
            [
              {
                "color": "#ffffff",
                "fontSize": 32,
              },
              [
                {
                  "lineHeight": 32,
                  "transform": [
                    {
                      "scaleX": 1,
                    },
                  ],
                },
                {
                  "backgroundColor": "transparent",
                },
              ],
              {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          󱢠
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`All components snapshot test: FavoriteCount snapshot: 1`] = `
<Text
  collapsable={false}
  numberOfLines={1}
  style={
    {
      "alignSelf": "flex-end",
      "backgroundColor": "rgba(103, 80, 164, 1)",
      "borderRadius": 10,
      "color": "rgba(255, 255, 255, 1)",
      "fontSize": 10,
      "height": 20,
      "lineHeight": 10,
      "minWidth": 20,
      "opacity": 1,
      "overflow": "hidden",
      "paddingHorizontal": 3,
      "textAlign": "center",
      "textAlignVertical": "center",
    }
  }
>
  0
</Text>
`;

exports[`All components snapshot test: MenuButton snapshot: 1`] = `
<View
  style={{}}
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "end": undefined,
        "flex": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "start": undefined,
        "top": undefined,
      }
    }
    testID="icon-button-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "flex": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="icon-button-container-inner-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": undefined,
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": undefined,
            "height": 40,
            "margin": 6,
            "overflow": "hidden",
            "width": 40,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityTraits="button"
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            [
              {
                "overflow": "hidden",
              },
              false,
              [
                {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                {
                  "borderRadius": 20,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              [
                {
                  "color": "rgba(73, 69, 79, 1)",
                  "fontSize": 24,
                },
                [
                  {
                    "lineHeight": 24,
                    "transform": [
                      {
                        "scaleX": 1,
                      },
                    ],
                  },
                  {
                    "backgroundColor": "transparent",
                  },
                ],
                {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            󰍜
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`All components snapshot test: SearchBarComponent snapshot: 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
      "paddingBottom": 16,
    }
  }
>
  <View
    collapsable={false}
    style={
      {
        "alignSelf": undefined,
        "bottom": undefined,
        "end": undefined,
        "flex": undefined,
        "left": undefined,
        "position": undefined,
        "right": undefined,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
        "start": undefined,
        "top": undefined,
      }
    }
    testID="search-bar-container-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "flex": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="search-bar-container-inner-layer"
    >
      <View
        collapsable={false}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#e0e0e0",
            "borderRadius": 200,
            "flex": undefined,
            "flexDirection": "row",
          }
        }
        testID="search-bar-container"
      >
        <View
          collapsable={false}
          style={
            {
              "alignSelf": undefined,
              "bottom": undefined,
              "end": undefined,
              "flex": undefined,
              "left": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
            }
          }
          testID="icon-button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "flex": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="icon-button-container-inner-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": undefined,
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "elevation": 0,
                  "flex": undefined,
                  "height": 40,
                  "margin": 6,
                  "overflow": "hidden",
                  "width": 40,
                }
              }
              testID="icon-button-container"
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel="search"
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityTraits="button"
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    false,
                    [
                      {
                        "alignItems": "center",
                        "flexGrow": 1,
                        "justifyContent": "center",
                      },
                      {
                        "borderRadius": 20,
                      },
                    ],
                  ]
                }
                testID="icon-button"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(28, 27, 31, 0.54)",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰍉
                </Text>
              </View>
            </View>
          </View>
        </View>
        <TextInput
          accessibilityRole="search"
          focusable={true}
          keyboardAppearance="light"
          onChangeText={[Function]}
          placeholder=""
          placeholderTextColor="rgba(28, 27, 31, 1)"
          returnKeyType="search"
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            [
              {
                "alignSelf": "stretch",
                "flex": 1,
                "fontSize": 18,
                "minWidth": 0,
                "paddingLeft": 8,
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {
                "marginLeft": -10,
              },
            ]
          }
          testID="search-bar"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          pointerEvents="none"
          testID="search-bar-icon-wrapper"
        >
          <View
            collapsable={false}
            style={
              {
                "alignSelf": undefined,
                "bottom": undefined,
                "end": undefined,
                "flex": undefined,
                "left": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "start": undefined,
                "top": undefined,
              }
            }
            testID="icon-button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "flex": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="icon-button-container-inner-layer"
            >
              <View
                collapsable={false}
                style={
                  {
                    "backgroundColor": undefined,
                    "borderColor": "rgba(121, 116, 126, 1)",
                    "borderRadius": 20,
                    "borderWidth": 0,
                    "elevation": 0,
                    "flex": undefined,
                    "height": 40,
                    "margin": 6,
                    "overflow": "hidden",
                    "width": 40,
                  }
                }
                testID="icon-button-container"
              >
                <View
                  accessibilityComponentType="button"
                  accessibilityLabel="clear"
                  accessibilityRole="button"
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": false,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityTraits="button"
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  centered={true}
                  collapsable={false}
                  focusable={true}
                  hitSlop={
                    {
                      "bottom": 6,
                      "left": 6,
                      "right": 6,
                      "top": 6,
                    }
                  }
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    [
                      {
                        "overflow": "hidden",
                      },
                      false,
                      [
                        {
                          "alignItems": "center",
                          "flexGrow": 1,
                          "justifyContent": "center",
                        },
                        {
                          "borderRadius": 20,
                        },
                      ],
                    ]
                  }
                  testID="icon-button"
                >
                  <Text
                    accessibilityElementsHidden={true}
                    allowFontScaling={false}
                    importantForAccessibility="no-hide-descendants"
                    pointerEvents="none"
                    selectable={false}
                    style={
                      [
                        {
                          "color": "rgba(255, 255, 255, 0)",
                          "fontSize": 24,
                        },
                        [
                          {
                            "lineHeight": 24,
                            "transform": [
                              {
                                "scaleX": 1,
                              },
                            ],
                          },
                          {
                            "backgroundColor": "transparent",
                          },
                        ],
                        {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        {},
                      ]
                    }
                  >
                    󰅖
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
